plugins/builtin/apm/datadog/plugin/plugin.go

```diff
@@ -0,0 +1,197 @@
+package plugin
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-api-client-go/api/v1/datadog"
+	hclog "github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/nomad-autoscaler/plugins"
+	"github.com/hashicorp/nomad-autoscaler/plugins/apm"
+	"github.com/hashicorp/nomad-autoscaler/plugins/base"
+)
+
+const (
+	// pluginName is the name of the plugin
+	pluginName = "datadog"
+
+	configKeyClientAPIKey = "dd_api_key"
+	configKeyClientAPPKey = "dd_app_key"
+
+	// naming convention according to datadog api
+	envKeyClientAPIKey = "DD_CLIENT_API_KEY"
+	envKeyClientAPPKey = "DD_CLIENT_APP_KEY"
```

Comment:
@urjitbhatia the changes look great, thanks so much for your work on this. I will run some local tests hopefully tomorrow and hope to get this merged in shortly after. A quick point on this section; is there are reason for having the client here, or can this be shortened in a similar way to the config map entries?

Reply:
You mean having these keys as constants? No particular reason - I can inline them if that works better with the code style.

Reply:
Apologies I wasn't clearer; let me suggest the code to make it easier.
```suggestion
	envKeyClientAPIKey = "DD_API_KEY"
	envKeyClientAPPKey = "DD_APP_KEY"
```

Reply:
Ah gotcha, that's how datadog API refers to those keys so I was trying to be consistent. Happy to merge this in, I don't think the word `CLIENT` adds much value.